### PR TITLE
`Adaptive learning`: Improve learning path management

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

This PR addresses improvements related to the management of Learning Paths in Artemis.
Stacked on #6747 .

### Description

- fixes several style issues
- fixes position of Learning Paths tab in `course management-tab-bar`
- adds endpoint to retrieve health status of learning paths in course
- adds endpoint to generate missing learning paths in course
- adds card in learning path management to display health status warning
- adds more information and navigation options in the details view of learning paths

### Steps for Testing

Prerequisites:
- 1 Admin
- Course (enrollment allowed)
- 1 Instructor
- \>2 Students (the more the merrier, 1 enrolled, 1 not enrolled yet)
- 1 Competency with 1 Exercise and 1 Lecture Unit

1. Log in to Artemis as Instructor
2. Navigate to Course Administration -> Learning Paths
3. You should see a message that learning paths are not enabled yet
4. Click the `Enable Learning Paths` button
5. The content should load the student's learning paths and display them as a pageable list
6. Make sure there is a Learning Path for each enrolled student
7. Try to search for students
8. Click on the button to view a student's Learning Path
9. A modal should open that displays the Learning Path Graph
10. Drag and Zoom the Graph and test if the Navigation-Buttons on top work as expected
11. Log in to Artemis as a unenrolled Student
12. Enroll in the course
13. Log in to Artemis as Instructor
14. Make sure the newly enrolled student's Learning Path is contained in the List
15. Manually add a user to the student group of your course
16. Make sure the student's learning path is now on the list under Learning Path Management

#### Performance Review
- [ ] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Screenshots

Learning Path Management:
Screenshot 2023-07-29 at 15 04 04

Progress/Details view:
Screenshot 2023-07-29 at 15 45 34

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/repository/LearningPathRepository.java [hunk 1]
```diff
@@ -72,4 +72,11 @@ default LearningPath findWithEagerCompetenciesAndLearningObjectsByIdElseThrow(lo
     default LearningPath findWithEagerCompetenciesAndLearningObjectsAndCompletedUsersByIdElseThrow(long learningPathId) {
         return findWithEagerCompetenciesAndLearningObjectsAndCompletedUsersById(learningPathId).orElseThrow(() -> new EntityNotFoundException("LearningPath", learningPathId));
     }
+
+    @Query("""
+            SELECT COUNT (learningPath)
+            FROM LearningPath learningPath
+            WHERE learningPath.course.id = :#{#courseId}
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/repository/LearningPathRepository.java:79]:
```suggestion
            WHERE learningPath.course.id = :courseId
```

File: src/main/java/de/tum/in/www1/artemis/service/LearningPathService.java [hunk 2]
```diff
@@ -410,4 +411,26 @@ public LearningObject getRecommendation(@NotNull LearningPath learningPath) {
...
+     * @param course the course for which the health status should be generated
+     * @return dto containing the health status and additional information (missing learning paths) if needed
+     */
+    public LearningPathHealthDTO getHealthStatusForCourse(Course course) {
+        if (!course.getLearningPathsEnabled()) {
+            return new LearningPathHealthDTO(LearningPathHealthDTO.HealthStatus.DISABLED);
+        }
+
+        var numberOfStudents = userRepository.countUserInGroup(course.getStudentGroupName());
+        var numberOfLearningPaths = learningPathRepository.countByCourseId(course.getId());
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/LearningPathService.java:427]: Explicit types would make it a bit clearer, why you compare with equals but then suddenly use a normal - for further calculations.
Why do these methods even return non primitive types?

File: src/main/webapp/i18n/de/competency.json [hunk 3]
```diff
@@ -153,15 +153,27 @@
...
                 "table": {
                     "name": "Name",
                     "login": "Login",
-                    "progress": "Fortschritt",
-                    "view": "Ansehen"
+                    "progress": "Fortschritt"
+                },
+                "progressNav": {
+                    "header": "Lernpfad",
+                    "refresh": "Refresh",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/competency.json:175]:
```suggestion
                    "refresh": "Aktualisieren",
```

File: src/main/webapp/i18n/de/competency.json [hunk 4]
```diff
@@ -153,15 +153,27 @@
...
+                    "center": "Center view"
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/competency.json:176]: What should the translation be here?

File: src/main/webapp/i18n/en/competency.json [hunk 5]
```diff
@@ -152,15 +152,27 @@
...
                     "name": "Name",
                     "login": "Login",
-                    "progress": "Progress",
-                    "view": "View"
+                    "progress": "Progress"
+                },
+                "progressNav": {
+                    "header": "Learning Path",
+                    "refresh": "Refresh",
+                    "center": "Center view"
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/en/competency.json:175]: What is a center view supposed to be?

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

File: src/main/webapp/i18n/en/competency.json (hunk 5 above)
Comment (MaximilianAnzinger) [reply src/main/webapp/i18n/en/competency.json:175]: The Graph of the learning path is right below this button. Users can zoom in and move around using their mouse and scroll wheel. If they want to zoom out and center the view to see the entire graph again, they can use this button.
I'm open for any suggestions to rename this hint 👍

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Tested on TS1 during the testing session:
- Why can I not enable learning paths in the course settings?
- The tooltip of "Enable learning paths" is not attached to its button
- The order in the navigation bar is different in the student view and instructor view. Learning Paths are before communication for students but after it in the instructor view
- The error message for Missing Learning paths still persists after generating the missing paths
- Some whitespace below the warning would be nice
- ...

File: src/main/java/de/tum/in/www1/artemis/repository/LearningPathRepository.java [hunk 6]
```diff
@@ -72,4 +72,11 @@ default LearningPath findWithEagerCompetenciesAndLearningObjectsByIdElseThrow(lo
...
+            """)
+    long countByCourseId(@Param("courseId") long courseId);
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/repository/LearningPathRepository.java:81]: I think you don't even need the Query here, JPA should be able to parse the name itself

File: src/main/webapp/i18n/en/competency.json [hunk 7]
```diff
@@ -152,15 +152,27 @@
             "learningPathButton": "Learning Paths",
             "manageLearningPaths": {
                 "title": "Learning Path Management",
+                "health": {
+                    "missing": {
+                        "title": "Missing Learning Paths",
+                        "body": "Some student's have not generated their Learning Paths yet. This is not critical. Their Learning Paths will be created once they request their learning path for the first time.",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/en/competency.json:158]:
```suggestion
                        "body": "Some student's have not generated their learning paths yet. This is not critical. Their learning paths will be created once they request their learning path for the first time.",
```

File: src/main/webapp/i18n/en/competency.json (hunk 5 above)
Comment (JohannesStoehr) [reply src/main/webapp/i18n/en/competency.json:175]: I think this In the context of the ui this is fine 👍

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Left some inline comments as well

**Comment by MaximilianAnzinger** [comment]
> Why can I not enable learning paths in the course settings?

Enabling learning paths prior to creating lecture units, exercises, and competencies does not make sense.

File: src/main/java/de/tum/in/www1/artemis/repository/LearningPathRepository.java (hunk 6 above)
Comment (MaximilianAnzinger) [reply src/main/java/de/tum/in/www1/artemis/repository/LearningPathRepository.java]: This query was actually the reason for the stuck health status warning. This is now resolved 👍

**Review by JohannesStoehr — APPROVED** [review]
Approve Code

**Pull request closed without merging**